lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -697,6 +697,84 @@ static LogicalResult legalizeHWModule(Block &block,
   // True if these operations are in a procedural region.
   bool isProceduralRegion = block.getParentOp()->hasTrait<ProceduralRegion>();
 
+  // Lower HW wires into SV wires in the appropriate spots. Try to put the
+  // declaration close to the first use in the block, and the assignment right
+  // after the assigned value becomes available.
+  struct WireLowering {
+    hw::WireOp wireOp;
+    Block::iterator declarePoint;
+    Block::iterator assignPoint;
+  };
+  SmallVector<WireLowering> wireLowerings;
+
+  for (auto hwWireOp : block.getOps<hw::WireOp>()) {
+    // Find the earliest point in the block at which the input operand is
```

Comment:
Just a general factoring note... this function has a somewhat interesting split of inline control flow versus what gets factored out into helpers. Keeping with that, maybe the body of this for loop could be factored into a helper like `buildWireLowerings`.

Reply:
Yeah great idea, let me factor this out a bit.

Reply:
👍 thanks for the refactoring, looks good to me.